acre/core.py

```diff
@@ -146,12 +191,12 @@ def parse(env, platform_name=None):
 
 
 def append(env, env_b):
-    """Append paths of environment b into environment"""
+    """Append paths of environment b into environment."""
     # todo: should this be refactored to "join" or "extend"
     # todo: this function name might also be confusing with "merge"
     env = env.copy()
     for variable, value in env_b.items():
-        for path in value.split(";"):
+        for path in value.split(os.pathsep):
```

Comment:
For what it's worth, I recall that originally the decision for `;` was intentional.

The environment definition as defined for acre in `.json` files were always supposed to be written with `;` as separator as can also be seen in the `parse()` function. Which you interestingly enough seem to have **not** patched up? Which I feel now can result in unexpected behavior.

Because on Linux `parse()` would then still create variables that can be joined with `;` (for values written as a list in the JSON, not those written as single string) but it will later try to split the value using `os.pathsep` in `append()`

Basically the intention is that the environments you are *appending* into the first are **always** written with the `;` separator.

Imagine this input:
```python
import acre
import os

data = {
    "MYVALUE": [
        "PATH1",
        "PATH2",
        "PATH3",
        "PATH4;PATH5"
    ]
}

env = acre.parse(data)
print env["MYVALUE"]
# "PATH1;PATH2;PATH3;PATH4;PATH5"
```
I believe even after your PR the result from that is the same. As such, would this actually merge into the environment the way you intended?

Could you give me a simple reproducable example of code that shows the issue you were facing? Potentially that single case could then also get its own test.

---

Note by the way that `lib.append_path` which was intended to append using the `os.pathsep` actually does. I do admit that the code seems a bit iffy on the multiplatform support, but even with these changes I still feel we're missing something that must be obvious.

The first decision to make is. **Do we always intend to use `;` as separator in the .json files, no matter the OS?** I'd vote yes, because you'd have to pick one standard so that the same variables "potentially" could be cross-platform, like writing `{MAYA_LOCATION};{MAYA_HOME}` in the input data.

In that case we'd need to make sure that no matter the OS that we always consider the `;` as the separator to split in the input environments.

Then, the only moment we need to ensure things are `:` is when they are intended for the current OS, basically 'translating' them into the local `os.pathsep`.

Unfortunately I've to jump into some other things today. But hopefully these notes help to give an idea of a possible solution for you to set up.